app/proxy_info.py

```diff
@@ -0,0 +1,64 @@
+import yaml
+from pathlib import Path
+
+from configs import config as cfg
+import log_util
+
+logger = log_util.get_logger(
+    __name__, cfg.LOG_LEVEL_TABLE.get(__name__, cfg.DEFAULT_LOG_LEVEL)
+)
+
+"""
+proxy_info.yaml:
+# version 1
+gateway: bool
+enable_ota_proxy: bool
+upper_ota_proxy: str
+generic_proxy: str
+local_server: Tuple[str, int]
+"""
+
+DEFUALT_PROXY_INFO = """
+gateway: false
+enable_ota_proxy: false
+"""
+
+
+class ProxyInfo:
+    def __init__(self, proxy_info_file: str = cfg.PROXY_INFO_FILE):
+        proxy_info_file_path = Path(proxy_info_file)
+        if proxy_info_file_path.is_file():
+            proxy_info: dict = yaml.safe_load(proxy_info_file_path.read_text())
+        else:
+            proxy_info: dict = yaml.safe_load(DEFUALT_PROXY_INFO)
+
+        self.enable_local_ota_proxy: bool = proxy_info.get("enable_ota_proxy", False)
+        self.upper_ota_proxy: str = proxy_info.get("upper_ota_proxy")
+        self.generic_proxy: str = proxy_info.get("generic_proxy")
+
+        if self.enable_local_ota_proxy:
+            self.gateway: bool = proxy_info.get("gateway", False)
+            self.host, self.port = proxy_info.get("local_server", ("0.0.0.0", 8000))
+
+    def get_proxy_for_local_ota(self) -> str:
+        if self.enable_local_ota_proxy:
+            # if local proxy is enabled, local ota client also uses it
+            return f"http://{self.host}:{self.port}"
+        elif self.upper_ota_proxy:
```

Comment:
Question: What is the difference between enable_local_ota_proxy and upper_ota_proxy?
Do we need a different configuration?
It means `upper_ota_proxy: localhost:8080` doesn't work?

Reply:
the `enable_local_proxy` will tell the ota_client to set up a local ota_proxy server, that can be used by local ota update and its child ecu update.
`upper_ota_proxy` is the parent ecu's ota_proxy server.
when both fields are set, the local_proxy server will use upper_ota_proxy as upper proxy(multi layer proxy).
If only `enable_local_proxy` is set, the local_proxy server will directly connect to the Internet by itself.

Reply:
What I think is that, if ota_proxy is enable, the local ota_client will also use it to download ota files, like its child ecus do.

Reply:
enable_local_proxy is required to launch a proxy server. understood.